src/frequenz/sdk/timeseries/_base_types.py

```diff
@@ -134,6 +134,39 @@ def map(
         )
 
 
+@runtime_checkable
+class Comparable(Protocol):
+    """
+    A protocol that requires the implementation of comparison methods.
+
+    This protocol is used to ensure that types can be compared using
+    the less than or equal to (`<=`) and greater than or equal to (`>=`)
+    operators.
+    """
+
+    def __le__(self, other: Any, /) -> bool:
+        """
+        Return True if self is less than or equal to other, otherwise False.
+
+        Args:
+            other: The value to compare with.
+
+        Returns:
+            bool: True if self is less than or equal to other, otherwise False.
+        """
+
+    def __ge__(self, other: Any, /) -> bool:
+        """
+        Return True if self is greater than or equal to other, otherwise False.
+
+        Args:
+            other: The value to compare with.
+
+        Returns:
+            bool: True if self is greater than or equal to other, otherwise False.
+        """
+
+
 _T = TypeVar("_T")
```

Comment:
My suggestion is to bind this `_T` to `Comparable | None`.
```suggestion
_T = TypeVar("_T", bound=Comparable | None)
```